geomesa-convert/geomesa-convert-parquet/src/main/scala/org/locationtech/geomesa/convert/parquet/ParquetConverterFactory.scala

```diff
@@ -109,6 +105,40 @@ class ParquetConverterFactory
       case NonFatal(e) => logger.debug(s"Could not infer Parquet converter from input:", e); None
     }
   }
+
+  private def computeTransformFunction(name: String, path: String, descriptor: AttributeDescriptor, schemaVersion: Int): String = {
+    // Make both expression vals lazy so that we only compute the one we need to, once we determine what the schema version is
+    lazy val expressionV2 = ObjectType.selectType(descriptor) match {
```

Comment:
can we make these functions instead of lazy vals? you can define functions inside your other functions, if you don't want them to be top-level. using functions seems a little clearer, and there is some compute overhead with initializing lazy vals.